tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -401,71 +479,74 @@ public void failCreateOnMismatchedDimensions() {
   }
 
   @Test
-  public void failCopyToOnIncompatibleDestination() {
+  public void whenCopyToWithDimensionMismatchThenThrowExpectedException() {
```

Comment:
`ThenThrowExpectedException` seems like a verbose way of saying `fail` :). I suppose there are other ways to fail. Anyway, the "when", "then", and "expected" words in all these tests do seem a bit verbose to me. I'd prefer keeping the names concise. For example, in this case `failCopyToOnDimensionMismatch` vs. `whenCopyToWithDimensionMismatchThenThrowExpectedException` - both of them seem like they convey the same information when shown up in a failure report.